src/components/ItemList.svelte

```diff
@@ -9,31 +9,45 @@
   import FilterInput from "./FilterInput.svelte";
   import Markdown from "./Markdown.svelte";
 
+  import { isExpired } from "../state/metrics";
+
   let DEFAULT_ITEMS_PER_PAGE = 20;
 
   export let appName;
   export let items;
   export let itemType;
-  export let filteredItems;
-  export let totalItems;
-  export let paginatedItems;
+
+  export let filterText = "";
+
+  let showExpired = false;
+  let filteredItems = items.filter((item) => !isExpired(item.expires));
+  let pagedItems;
 
   let currentPage = writable(1);
   setContext("currentPage", currentPage);
 
-  totalItems = items.length;
-  paginatedItems = chunk([...items], DEFAULT_ITEMS_PER_PAGE);
+  const goToPage = (page) => {
+    pagedItems =
+      filteredItems.length > 0
+        ? chunk([...filteredItems], DEFAULT_ITEMS_PER_PAGE)[page - 1]
+        : [];
+  };
 
-  function filterItems(filterText) {
-    filteredItems = items.filter((item) => item.name.includes(filterText));
-    if (filteredItems.length > 0) {
-      currentPage.set(1);
-      paginatedItems = chunk([...filteredItems], DEFAULT_ITEMS_PER_PAGE);
-      totalItems = filteredItems.length;
-    }
-  }
+  const handleFilter = () => {
+    let shownItems = showExpired
+      ? items
+      : items.filter((item) => !isExpired(item.expires));
+    filteredItems = shownItems.filter((item) => item.name.includes(filterText));
+    // show the first page of result
+    currentPage.set(1);
+    goToPage(1);
```

Comment:
Is this necessary? I feel like if we set the store we should get the callback automatically below.

Reply:
if `currentPage` is already 1, setting it to 1 would not call `goToPage()`

Reply:
Makes sense, maybe we could just add a quick comment about that?